ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/show/ShowPartitionsOperation.java

```diff
@@ -54,12 +55,17 @@ public ShowPartitionsOperation(DDLOperationContext context, ShowPartitionsDesc d
   @Override
   public int execute() throws HiveException {
     Table tbl = context.getDb().getTable(desc.getTabName());
-    if (!tbl.isPartitioned()) {
-      throw new HiveException(ErrorMsg.TABLE_NOT_PARTITIONED, desc.getTabName());
+    boolean isIcebergTable = isIcebergTable(tbl);
+    if (!(tbl.isPartitioned() || isIcebergTable)) {
+        throw new HiveException(ErrorMsg.TABLE_NOT_PARTITIONED, desc.getTabName());
```

Comment:
Can we generalize this part rather than hardcoding ICEBERG? I guess this is useful for another storage handler.

Reply:
Haha. We've been developing a tailored storage with some similar concepts to Iceberg for 10 years. We really like recent changes for Iceberg because it would give us a chance to move our hacks to standard APIs of storage handlers. Let me check newly added APIs and propose better generalization.

Reply:
this block is to handle cases when you call `show partitions` on non-partitioned table. no need to protect if there is no partition support especially when you added the default impl for showPartitions in storage handler